src/Tests/Microsoft.DotNet.GenAPI.Tests/CSharpFileBuilderTests.cs

```diff
@@ -1284,6 +1284,43 @@ public partial class C : B
                     """);
         }
 
+        [Fact]
+        public void TestBaseTypeWithAmbiguousNonDefaultConstructors()
+        {
+            RunTest(original: """
+                    namespace Foo
+                    {
+                        public class A
+                        {
+                            public A(char c) { }
+                            public A(int i) { }
+                            public A(string s) { }
+                        }
+
+                        public class B : A
+                        {
+                            public B() : base("") {}
+                        }
+                    }
+                    """,
+                expected: """
+                    namespace Foo
+                    {
+                        public partial class A
+                        {
+                            public A(char c) { }
+                            public A(int i) { }
+                            public A(string s) { }
+                        }
+
+                        public partial class B : A
+                        {
+                            public B() : base(default(char)) {}
+                        }
+                    }
+                    """);
+        }
+
```

Comment:
@ericstj do you mind also adding test cases for the examples provided in the issue? https://github.com/dotnet/sdk/issues/31655

One default is casted to a class type, the other one to a generic type:

```cs
base((Packaging.Core.PackageIdentity)default!, (System.Collections.Generic.IEnumerable<Packaging.Core.PackageDependency>)default!)
```

Want to make sure both work as expected.

Reply:
Sure I can, though I don't think we need to use those NuGet types.  I'll add some user types.

Reply:
Interesting, so I'm adding the type arguments to default correctly, but it's being removed by https://github.com/dotnet/sdk/blob/4ea3e8304c21022bbea72a29154b7602af4f9239/src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs#L94

This seems to be a bug in the Reducer in roslyn.

Reply:
Filed https://github.com/dotnet/roslyn/issues/67867 and I will add the regression test with a suppression.